.github/workflows/test.yaml

```diff
@@ -1,36 +1,115 @@
 # /.github/workflows/test.yaml
-# Build, lint and test the library
+# Test everything with the library
 
 name: Test
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
 
 jobs:
-  test:
-    name: Build, Lint and Test
+  build_library:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+      - name: Use Node 16
+        uses: actions/setup-node@v1
+        with:
+          node-version: 16
+      - name: Build the library
+        run: |
+          npm ci
+          npm pack
+      - name: Use the built library to run extra tests
+        run: |
+          mv ./express-rate-limit*.tgz ../express-rate-limit.tgz
+          git fetch --no-tags --depth=1 origin tests
+          git checkout tests
+          cp ../express-rate-limit.tgz ./assets/
+
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+          git add assets/
+          git commit --allow-empty --message "chore: update library tarball"
+          git push
+  test_library:
+    name: Test
     strategy:
       matrix:
         node-version: [12.x, 14.x, 15.x, 16.x, 17.x]
         os: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
-
     steps:
-      # This is required to ensure the linter does not fail on Windows
-      - name: Disable `git autocrlf`
-        run: |
-          git config --global core.autocrlf false
       - name: Checkout the repository
         uses: actions/checkout@v2
       - name: Use Node ${{ matrix.node-version }} on ${{ matrix.os }}
         uses: actions/setup-node@v1
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Install dependencies
-        run: npm ci
-      - name: Build, lint and test the library
-        run: npm test
-      - name: Upload coverage report
-        uses: actions/upload-artifact@v2
+      - name: Test the library
+        run: |
+          npm ci
+          npm test
+  test_imports:
+    name: Import Variations
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 15.x, 16.x, 17.x]
+    runs-on: ubuntu-latest
+    needs: build_library
+    steps:
+      - name: Checkout the `tests` branch
+        uses: actions/checkout@v2
         with:
-          name: Coverage Report
-          path: coverage/lcov-report
+          ref: tests
+      - name: Checkout the repository
+        uses: actions/checkout@v2
+      - name: Use Node ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Test various imports with the library
+        run: |
+          git checkout origin/tests
+          cd test/import-tests
+          bash run-all-tests
```

Comment:
Why not include these tests in the main branch?

Reply:
Sure, I'll work on that

Reply:
Done, but need to make the external store tests use jest as a test runner. Also, switched the tests to run on only ubuntu, as running docker services is not supported on windows and macos 😕

Reply:
All tests use jest now! 🚀